nemo_curator/modules/splitter.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Optional
+
+import pandas as pd
+
+from nemo_curator.datasets import DocumentDataset
+
+
+class DocumentSplitter:
```

Comment:
Should import from BaseModule ? 

```suggestion
class DocumentSplitter(BaseModule):
```

Reply:
This should work with both `cuDF` and `pandas`.

Reply:
Thanks. Fixed this and added cudf tests too.